### PR TITLE
Show subspace events in Space

### PR DESCRIPTION
**Pull request by hero101**
tba

## Release Notes

- **New Features**
	- Introduced utility functions: `convertToEntity`, `getDiff`, and `hasOnlyAllowedFields` for enhanced entity management and validation.
	- Added `allowEventsFromSubspaces` property to collaboration settings across various space configurations, enabling event creation from subspaces.
	- New fields in event-related data structures to manage visibility on parent calendars.

- **Bug Fixes**
	- Improved error handling and logic in space and event management services.

- **Documentation**
	- Updated documentation to reflect new functionalities and properties.

- **Tests**
	- Comprehensive test suites added for new utility functions and existing functionalities to ensure robustness.

### CODE REVIEW

File: src/common/utils/get-differences/get.diff.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * @returns {Partial<T> | null} - A partial object containing the differences, or null if no differences are found.
+ */
+export const getDiff = <T extends Record<string, any>>(
+  obj1: T,
+  obj2: T
+): Partial<T> | null => {
+  const result: Partial<T> = {};
+
+  for (const key in obj1) {
+    if (obj1.hasOwnProperty(key) && obj2.hasOwnProperty(key)) {
```
[the remark this answers is not in the document]
Comment (hero101) [reply src/common/utils/get-differences/get.diff.ts:18]: this function is not available to the project

File: src/domain/timeline/event/event.resolver.mutations.ts [hunk 2]
```diff
@@ -27,13 +27,13 @@ export class CalendarEventResolverMutations {
   ): Promise<ICalendarEvent> {
     const calendarEvent =
       await this.calendarEventService.getCalendarEventOrFail(deleteData.ID);
-    await this.authorizationService.grantAccessOrFail(
+    this.authorizationService.grantAccessOrFail(
       agentInfo,
       calendarEvent.authorization,
       AuthorizationPrivilege.DELETE,
       `delete calendarEvent: ${calendarEvent.id}`
     );
```
[the remark this answers is not in the document]
Comment (hero101) [reply src/domain/timeline/event/event.resolver.mutations.ts:35]: `grantAccessOrFail` is not async

File: src/migrations/1730464361986-allowEventsFromSubspaces.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class AllowEventsFromSubspaces1730464361986 implements MigrationInterface {
+    name = 'AllowEventsFromSubspaces1730464361986'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+      // add setting to calendar event to allow events from subspaces
+      // defaults to false per requirements
+      await queryRunner.query(`ALTER TABLE \`calendar_event\` ADD \`visibleOnParentCalendar\` tinyint NOT NULL`);
```
[the remark this answers is not in the document]
Comment (hero101) [reply src/migrations/1730464361986-allowEventsFromSubspaces.ts]: the default is 0

**Review by bobbykolev — DISMISSED** [review]
🤯

**Review by bobbykolev — APPROVED** [review]

**Review by bobbykolev — APPROVED** [review]

**Pull request merged**